### PR TITLE
Headless/tahsin/super admin/branch/crud

### PR DESCRIPTION
**Pull request by ghost**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [ ] Link to a **GH Issue** or **Jira** task here for reference.
- [ ] Bug fix
- [ ] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [x] Separation of Concerns followed (Reviewer name/NaN)
- [x] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [x] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [x] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/headless/src/database/mongodb/branch/branch.model.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            index: true
+        },
+        url: {
+            type: String,
+            unique: true
+        },
+        address: BranchAddressSchema,
+        isActive: {
+            type: Boolean,
+            default: false,
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/branch/branch.model.ts]: keep default as true

**Comment by ghost** [comment]
[https://pm23.atlassian.net/browse/BC-660](url)

File: packages/headless/src/database/mongodb/branch/branch.model.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        },
+    },
+    {
+        timestamps: true,
+        versionKey: false,
+    },
+);
+
+BranchSchema.index({ store: 1 });
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/branch/branch.model.ts]: already you've defined the index. no need to redefine

**Review by Anjarul-Islam — CHANGES_REQUESTED** [review]

File: packages/headless/src/database/mongodb/branch/branch.model.ts (hunk 1 above)
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/branch/branch.model.ts]: keep default as true

File: packages/headless/src/modules/branch/rest/branch.controller.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    Patch,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { ApiBearerAuth, ApiQuery, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { CreateBranchSuccessResponseDto, CreateBranchErrorResponseDto, CreateBranchRequestDto } from './dto/create.branch.dto';
+import { BranchService } from '../services';
+import { SingleBranchSuccessResponseDto, SingleBranchErrorResponseDto } from './dto/branch.dto';
+
+@ApiTags('Branch API')
+@Controller('branch')
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/branch/rest/branch.controller.ts]: rename the route as `store-branches`

File: packages/headless/src/modules/branch/rest/branch.controller.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        @Body() branch: CreateBranchRequestDto,
+        @Res({ passthrough: true }) res: Response,
+    ){
+        const { code, ...response} = await this.branchService.createBranch(branch);
+        
+        res.status(code)
+        return { code, ...response};
+    }
+
+    @Get('/allBranch/:storeId')
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/branch/rest/branch.controller.ts]: rename the route as `/:storeId` only

**Pull request merged**